src/librustc_resolve/build_reduced_graph.rs

```diff
@@ -391,6 +391,9 @@ impl<'a> Resolver<'a> {
                 self.define(parent, ident, TypeNS, (module, vis, sp, expansion));
                 self.current_module = module;
             }
+            ItemKind::TraitAlias(..) => {
+                // TODO: do we need to do anything here?
```

Comment:
Yes, the same thing as in `ItemKind::Ty` (only with `Def::TraitAlias`).

Reply:
You'll also have to add a new variant to `PathSource` for contexts permitting trait alias paths (they are slightly different than contexts permitting trait paths.)

Reply:
Why would the contexts be different? Because `impl Alias {}` won't be allowed?

Reply:
Yes.
So you'll automatically get "expected a trait, found a trait alias" errors on `impl Alias {}`.

Reply:
This already happens with no modifications to `PathSource`.

Reply:
Oh but maybe this is needed to check for `impl Alias for Struct {}`?